### DISCUSSION TITLE
LoCon

### DISCUSSION THREAD

**Discussion by bmaltais** [bmaltais/kohya_ss#260] [General]
Starting a discussion about LoCon vs LoRA. I intend to merge a PR into dev -> master for LoCon support.

You can discuss this new type of LoRA here.

**Comment by bmaltais** [comment]
LoCon are now supported in the master branch. Just select the LoCon LoRA type in the Dreambooth LoRA tab:

![image](https://user-images.githubusercontent.com/7474674/222534724-22154e8e-19fc-42e1-a919-c1f3983a0423.png)

It will then make those 2 new parameters visible when selected:

![image](https://user-images.githubusercontent.com/7474674/222534821-11e9aac5-0bc6-40ca-a6a8-65266dbb41f6.png)

If using LoCon you can try to cut the original LoRA rank in half and make the LoCon conv rank the same as the LoRA rank as a 1st start.

**Comment by phoeaven** [comment]
I tried various combination of rank/alpha and conv rank/alpha with different optimizers. It seems that LoCon is more likely to reproduce the character I trained, just like those LoRAs that trained without caption. For example, even short prompts could generate more accurate character. But some small things like hair ornament cannot be controlled easily with prompts and negative prompts.

**Reply by bmaltais** [reply]
Interesting observation. LoCon is probably not the ultimate tool... but one that can help in some cases.

**Comment by slide1980** [comment]
Any info about LoCon usage in Lora training and applying models in SD, please. What will the use of **Composable LoRA**?
https://github.com/opparco/stable-diffusion-webui-composable-lora

**Reply by bmaltais** [reply]
I have a feeling LoCon might not work with compostable LoRA. Better stick with standard LoRA for now.

**Comment by FurkanGozukara** [comment]
@bmaltais which one is best atm?

Kohya DyLoRA , Kohya LoCon , LyCORIS/LoCon , LyCORIS/LoHa , Standard

and each one requires different rank vector dimension and learning rate?
which one do you suggest for teaching a person or a style?

i am working on a tutorial video

by the way torch 1.13 works faster than torch 2 on rtx 3090 for lora training is that expected? with latest cudnn and xformers
